example/example.pro

```diff
@@ -9,17 +9,23 @@ DEFINES += SRCDIR=\\\"$$PWD/\\\"
 }
 
 # depends SqliteMigrator.lib {
-CONFIG(release, debug|release): LIBS += -lSqliteMigrator
-else:CONFIG(debug, debug|release): LIBS += -lSqliteMigratord
+CONFIG(release, debug|release): LIBS += -lSqliteMigrator -lQSqlMigrator
+else:CONFIG(debug, debug|release): LIBS += -lSqliteMigratord -lQSqlMigrator
 
 INCLUDEPATH += $$QSQLMIGRATOR_ROOT/plugins
 DEPENDPATH += $$DESTDIR
 
-win32:CONFIG(release, debug|release): PRE_TARGETDEPS += $$LIB_PATH/SqliteMigrator.dll
-else:win32:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$LIB_PATH/SqliteMigratord.dll
-else:unix:CONFIG(release, debug|release): PRE_TARGETDEPS += $$LIB_PATH/libSqliteMigrator.so
-else:unix:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$LIB_PATH/libSqliteMigratord.so
-# }
+static {
+    win32:CONFIG(release, debug|release): PRE_TARGETDEPS += $$LIB_PATH/QSqlMigrator.lib
+    else:win32:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$LIB_PATH/QSqlMigratord.lib
+    else:unix:CONFIG(release, debug|release): PRE_TARGETDEPS += $$LIB_PATH/libQSqlMigrator.a
+    else:unix:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$LIB_PATH/libQSqlMigratord.a
+} else {
+    win32:CONFIG(release, debug|release): PRE_TARGETDEPS += $$LIB_PATH/QSqlMigrator.dll
+    else:win32:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$LIB_PATH/QSqlMigratord.dll
+    else:unix:CONFIG(release, debug|release): PRE_TARGETDEPS += $$LIB_PATH/libQSqlMigrator.so
+    else:unix:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$LIB_PATH/libQSqlMigratord.so
+}
```

Comment:
It feels a bit odd that you remove `/libSqliteMigrator.so` and add `/libQSqlMigrator.so` for the shared build.

If I remember correctly, these `PRE_TARGETDEPS` are not required for a single build. They are used to consider if this library is required to link again, when a library changed. So I guess this breaks incremental builds.

Reply:
Thanks for quick reply.

Sorry, my bad, it was a copy-paste mistake. Fixed here and in other places.

Just checked: if I change some file in plugins/SqliteMigrator, the example and other targets are successfully rebuilt.